File: eventual-consistency-support-core/src/main/java/org/xyattic/eventual/consistency/support/core/consumer/aop/MessageProvider.java
package org.xyattic.eventual.consistency.support.core.consumer.aop;

import org.aspectj.lang.ProceedingJoinPoint;

import java.io.Serializable;

/**
 * @author wangxing
 * @create 2020/8/24
 */
public interface MessageProvider<T extends Serializable> {

    T getMessage(ConsumeMqMessage consumeMqMessage, Object[] args);

    default T getMessage(ConsumeMqMessage consumeMqMessage, ProceedingJoinPoint pjp) {
        return getMessage(consumeMqMessage, pjp.getArgs());
    }

}

File: eventual-consistency-support-core/src/main/java/org/xyattic/eventual/consistency/support/core/consumer/aop/MessageProvider.kt
package org.xyattic.eventual.consistency.support.core.consumer.aop

import org.aspectj.lang.ProceedingJoinPoint
import org.xyattic.eventual.consistency.support.core.consumer.AbstractMessage

/**
 * @author wangxing
 * @create 2020/8/24
 */
interface MessageProvider<T : AbstractMessage?> {

    fun getMessage(consumeMqMessage: ConsumeMqMessage?, args: Array<Any?>?): T

    fun getMessage(consumeMqMessage: ConsumeMqMessage?, pjp: ProceedingJoinPoint): T {
        return getMessage(consumeMqMessage, pjp.args)
    }

}